src/main/scala/com/microsoft/ml/spark/lightgbm/LightGBMBase.scala

```diff
@@ -144,30 +144,50 @@ trait LightGBMBase[TrainedModel <: Model[TrainedModel]] extends Estimator[Traine
     }
   }
 
+  /**
+    * Retrieves the categorical indexes in the features column.
+    * @param df The dataset to train on.
+    * @return the categorical indexes in the features column.
+    */
+  private def getCategoricalIndexes(df: DataFrame): Array[Int] = {
```

Comment:
there seems to be some merge conflict here?  Maybe your master branch is not updated to latest?